addons/account_asset/models/account_invoice.py

```diff
@@ -108,3 +108,9 @@ def _set_additional_fields(self, invoice):
             elif invoice.type == 'in_invoice':
                 self.asset_category_id = self.product_id.product_tmpl_id.asset_category_id.id
         super(AccountInvoiceLine, self)._set_additional_fields(invoice)
+
+    @api.v8
+    def get_invoice_line_account(self, type, product, fpos, company):
+        if product.asset_category_id and product.asset_category_id.account_asset_id:
+            return product.asset_category_id.account_asset_id
+        return super(AccountInvoiceLine, self).get_invoice_line_account(type, product, fpos, company)
```

Comment:
`return product.asset_category_id.account_asset_id or super(AccountInvoiceLine, self).get_invoice_line_account(type, product, fpos, company)` is probably enough